src/component_updater.py

```diff
@@ -0,0 +1,133 @@
+import logging
+import os
+from utils import io
+from atmos_component import AtmosComponent, COMPONENT_YAML
+import subprocess
+from github_provider import GitHubProvider
+from git_provider import GitProvider
+
+COMMIT_MESSAGE_TEMPLATE = "Updated component '{component_name}' to version '{component_version}'"
+
+
+class ComponentUpdater:
+    def __init__(self, github_provider: GitHubProvider, infra_repo_dir: str, new_version: str):
+        self.__github_provider = github_provider
+        self.__infra_repo_dir = infra_repo_dir
+        self.__new_version = new_version
+
+    def update(self, component_file: str):
+        original_component = AtmosComponent(self.__infra_repo_dir, component_file)
+
+        logging.info(f"Processing component: {original_component.get_name()}")
+
+        if self.__is_already_updated(original_component):
+            logging.info(f"Component already updated: {original_component.get_name()}")
+            return
+
+        updated_component = self.__clone_infra_for_component(original_component)
+
+        if self.__has_component_not_been_pulled(updated_component):
+            logging.debug(f"Component was not pulled: {original_component.get_name()}")
+            updated_component.update_version(self.__new_version)
+            updated_component.persist()
+            self.__update_component(original_component, updated_component)
+            return
+
+        rendered_component = self.__clone_infra_for_component(original_component)
+        rendered_component.update_version(self.__new_version)
+        io.remove_all_from_dir(rendered_component.get_component_dir())
+        rendered_component.persist()
+
+        self.__execute_atmos_vendor_pull(rendered_component.get_infra_repo_dir(), rendered_component.get_name())
+
+        rendered_component_files = io.get_filenames_in_dir(rendered_component.get_component_dir(), ['**/*'])
```

Comment:
Why is it a "rendered component"? It's just downloaded, no?
```suggestion
        rendered_component_files = io.get_filenames_in_dir(rendered_component.get_component_dir(), ['**/*'])
```

Reply:
Or rather "vendored_component"

Reply:
make sense, will rename